### PR TITLE
Update Finagle to 21.4.0

### PR DESCRIPTION
**Pull request by hmhagberg**
Update Finagle from 20.4.1 to 21.4.0. Also update following dependencies
to match versions used by Finagle:

* netty 4.1.47 -> 41.1.59
* netty-tcnative-boringssl-static 2.0.30 -> 2.0.35
* jackson 2.9.10 -> 2.11.2

Signed-off-by: Henri Hagberg <henri.hagberg@relexsolutions.com>

### CODE REVIEW

File: project/Base.scala [hunk 1]
```diff
@@ -163,6 +163,7 @@ class Base extends Build {
       case "BUILD" => MergeStrategy.discard
       case "com/twitter/common/args/apt/cmdline.arg.info.txt.1" => MergeStrategy.discard
       case "META-INF/io.netty.versions.properties" => MergeStrategy.last
+      case "module-info.class" => MergeStrategy.discard
```
Comment (hmhagberg) [inline project/Base.scala:166]: module-info.class was introduced in Jackson 2.10. Discarding it should be safe

File: router/core/src/main/scala/io/buoyant/router/Router.scala [hunk 2]
```diff
@@ -313,13 +313,20 @@ object StackRouter {
      * Augment the default client StatsFilter with a
      * per-logical-destination stats filter.
      */
-    def mkStack[Req, Rsp](orig: Stack[ServiceFactory[Req, Rsp]]): Stack[ServiceFactory[Req, Rsp]] =
-      orig.insertBefore(StackClient.Role.protoTracing, ClassifiedTracing.module[Req, Rsp])
-        .insertBefore(StatsFilter.role, PerDstPathStatsFilter.module[Req, Rsp])
+    def mkStack[Req, Rsp](orig: Stack[ServiceFactory[Req, Rsp]]): Stack[ServiceFactory[Req, Rsp]] = {
+      // Finagle replaces protoTracing with HttpTracing for HTTP(2) clients
```
Comment (hmhagberg) [inline router/core/src/main/scala/io/buoyant/router/Router.scala:317]: https://github.com/twitter/finagle/commit/78d93fde6cdf18534d6990ec7c8d4e97260e5c57

File: telemetry/zipkin/src/main/scala/com/twitter/finagle/buoyant/zipkin/ZipkinTracer.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      .withSessionQualifier.noFailFast
+      .withTracer(NullTracer)
+      .newService(Name.bound(Address(host, port)), "zipkin-tracer")
+
+    val client: Scribe.MethodPerEndpoint = new Scribe.FinagledClient(
+      new TracelessFilter andThen transport,
+      Protocols.binaryFactory()
+    )
+
+    val rawTracer = ScribeRawZipkinTracer(
```
Comment (hmhagberg) [inline telemetry/zipkin/src/main/scala/com/twitter/finagle/buoyant/zipkin/ZipkinTracer.scala:40]: This factory method was made private to finagle in https://github.com/twitter/finagle/commit/7723a949eae3efa0d9a256a816712716e4f10ce5. Copying this whole `ZipkinTracer` to a finagle package is not very ideal solution but I did not see any other way of supplying our own client.

**Comment by hmhagberg** [comment]
Ping @adleong @olix0r @zaharidichev, could I get review for this? Thanks!

**Comment by hmhagberg** [comment]
Test failure seems to happen in master as well https://app.circleci.com/pipelines/github/linkerd/linkerd/853/workflows/3e2f1f84-6c4a-4d29-8176-c1797190925e/jobs/10684

**Comment by cpretzer** [comment]
@adleong @olix0r @zaharidichev 

Can one of you approve this? I'll work on an RC build once it's merged

**Review by adleong — APPROVED** [review]

**Comment by cpretzer** [comment]
hi @hmhagberg just want to let you know that these changes have been pulled in and [1.7.5-rc1](https://github.com/linkerd/linkerd/releases/tag/1.7.5-rc1) is ready for you to test 😄

**Comment by hmhagberg** [comment]
Thanks!

**Pull request merged**